lib/puppet/resource_api.rb

```diff
@@ -304,6 +290,8 @@ def call_provider(value); end
                           my_provider.get(context).find { |h| namevar_match?(h) }
                         end
 
+        @warning_count = 0 # always reset when called
```

Comment:
Don't reset here. After 100 warnings for a type, there is no additional info that devs can get from this, and we don't want to kill production if someone slips in hitting this.